contrib/ansible/roles/cluster_operator_certgen/tasks/main.yml

```diff
@@ -0,0 +1,54 @@
+---
+- name: check if CA needs to be generated
+  stat:
+    path: "{{ ca_file }}"
+  register: ca_stat
+
+- name: write CA setup json
+  template:
+    src: ca-setup.json
+    dest: "{{ ca_setup_file }}"
+  when: not ca_stat.stat.exists
+
+- name: write CA config json
+  template:
+    src: ca-config.json
+    dest: "{{ ca_config_file }}"
+  when: not ca_stat.stat.exists
+
+- name: generate CA
+  shell: "cfssl genkey --initca {{ ca_setup_file }} | cfssljson -bare {{ ca_name }}"
+  args:
+    chdir: "{{ cert_dir }}"
+  when: not ca_stat.stat.exists
+
+- name: check if cert needs to be generated
+  stat:
+    path: "{{ cert_file }}"
+  register: cert_stat
+
+- name: write cert config json
+  template:
+    src: cert-config.json
+    dest: "{{ cert_config_file }}"
+  when: not cert_stat.stat.exists
+
+- name: generate cert
+  shell: "cfssl gencert -ca={{ ca_file }} -ca-key={{ ca_key_file }} -config={{ ca_config_file }} {{ cert_config_file }} | cfssljson -bare {{ cert_name }}"
+  args:
+    chdir: "{{ cert_dir }}"
+  when: not cert_stat.stat.exists
+
+#- name: cleanup temporary files
```

Comment:
Is there a reason you commented out this part?

Reply:
Just debugging so I could diff these with the old variants, actually exposed the problem. Readding the cleanup next commit.